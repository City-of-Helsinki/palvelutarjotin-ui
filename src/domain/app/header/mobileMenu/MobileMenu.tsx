import classNames from 'classnames';
import React from 'react';

import { SUPPORTED_LANGUAGES } from '../../../../constants';
import useLocale from '../../../../hooks/useLocale';
import { i18n, useTranslation } from '../../../../i18n';
import { Language } from '../../../../types';
import styles from './mobileMenu.module.scss';

export const testIds = {
  menu: 'mobile-menu',
};

interface Props {
  id?: string;
  isMenuOpen: boolean;
  onClose: () => void;
}

const MobileMenu: React.FC<Props> = ({ id, isMenuOpen, onClose }) => {
  const { t } = useTranslation();
  const locale = useLocale();

  const changeLanguage = (newLanguage: Language) => {
    i18n.changeLanguage(newLanguage);
    onClose();
  };

  return (
    <div
      aria-hidden={!isMenuOpen}
      className={classNames(styles.mobileMenu, {
        [styles.menuOpen]: isMenuOpen,
      })}
      data-testid={testIds.menu}
      id={id}
      style={{ visibility: isMenuOpen ? undefined : 'hidden' }}
    >
      <div className={styles.linksWrapper}></div>
      <div className={styles.languageSelect}>
        <ul>
          {Object.values(SUPPORTED_LANGUAGES).map((language) => (
            <li
              key={language}
              className={classNames(styles.languageLink, {
                [styles.isSelected]: locale === language,
              })}
            >
              <button lang={language} onClick={() => changeLanguage(language)}>
                {t(`header:languages.${language}`)}
              </button>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default MobileMenu;
