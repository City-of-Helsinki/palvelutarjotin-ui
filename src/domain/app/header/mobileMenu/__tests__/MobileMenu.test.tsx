import { render } from '@testing-library/react';
import React from 'react';

import MobileMenu from '../MobileMenu';

it('MobileMenu matches snapshot', () => {
  const { container } = render(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <MobileMenu isMenuOpen={true} onClose={() => {}} />
  );

  expect(container.firstChild).toMatchSnapshot();
});
