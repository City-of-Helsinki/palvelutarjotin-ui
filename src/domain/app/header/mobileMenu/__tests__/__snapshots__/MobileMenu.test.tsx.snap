// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu matches snapshot 1`] = `
<div
  aria-hidden="false"
  class="mobileMenu menuOpen"
  data-testid="mobile-menu"
>
  <div
    class="linksWrapper"
  />
  <div
    class="languageSelect"
  >
    <ul>
      <li
        class="languageLink isSelected"
      >
        <button
          lang="fi"
        >
          Suomeksi
        </button>
      </li>
      <li
        class="languageLink"
      >
        <button
          lang="sv"
        >
          PÃ¥ svenska
        </button>
      </li>
      <li
        class="languageLink"
      >
        <button
          lang="en"
        >
          In English
        </button>
      </li>
    </ul>
  </div>
</div>
`;
